File: src/aws/lambdas/incremental_distribution/src/test/java/uk/nhs/nhsx/core/TestEnvironments.java
package uk.nhs.nhsx.core;

import java.util.Map;
import java.util.Optional;
import java.util.function.Function;

public class TestEnvironments {
    public static final Environment.Access NOTHING = new Environment.Access() {
        @Override
        public <T> Optional<T> optional(Environment.EnvironmentKey<T> key) {
            return Optional.empty();
        }
    };
    
    public static final Environment EMPTY = Environment.fromName("empty", NOTHING);

    public static final Function<Map<String, String>, Environment> TEST = t -> Environment.fromName("test", Environment.Access.TEST.apply(t));
}

File: src/aws/lambdas/incremental_distribution/src/test/java/uk/nhs/nhsx/core/TestEnvironments.kt
package uk.nhs.nhsx.core

import uk.nhs.nhsx.core.Environment.EnvironmentKey
import java.util.*
import java.util.function.Function

object TestEnvironments {

    val NOTHING: Environment.Access = object : Environment.Access {
        override fun <T> optional(key: EnvironmentKey<T>): Optional<T> = Optional.empty()
    }

    val TEST = Function { t: Map<String, String> -> Environment.fromName("test", Environment.Access.TEST.apply(t)) }

    fun environmentWith(vararg pairs: Pair<String, String>): Environment = TEST.apply(mapOf(*pairs.toList().toTypedArray()))
}
